src/Http/Http/src/Features/FormFeature.cs

```diff
@@ -182,9 +184,11 @@ private async Task<IFormCollection> InnerReadFormAsync(CancellationToken cancell
                             var fileSection = new FileMultipartSection(section, contentDisposition);
 
                             // Enable buffering for the file if not already done for the full body
+                            var httpBufferingOptions = _request.HttpContext.RequestServices.GetService<IOptions<HttpBufferingOptions>>();
```

Comment:
```suggestion
                            var httpBufferingOptions = _request.HttpContext.RequestServices.GetRequiredService<IOptions<HttpBufferingOptions>>();
```

Reply:
If we are registering the factory I think here we should resolve it and pass it to the EnableRewind for the `MultipartSection` as an overload. We can let the original method creating the `FileBufferingReadStream` manually in case someone was using it, since it was public. Or just delete it given that we are keeping, at least for now, the use of the Environment Variable via the `PostConfigure`.